### ISSUE TITLE
Product with +1.000 variations and Slowly to Add to Cart

### ISSUE THREAD

**Issue by leomiranda92** [woocommerce/woocommerce#19587]
## Prerequisites

- [x] I have searched for similar issues in both open and closed tickets and cannot find a duplicate
- [x] The issue still exists against the latest `master` branch of WooCommerce on Github (this is **not** the same version as on WordPress.org!)
- [x] I have attempted to find the simplest possible steps to reproduce the issue
- [ ] I have included a failing test as a pull request (Optional)

## Steps to reproduce the issue

1. Create an Product with more than 1.000 variations
2. After selecting the options in product page and wc ajax get variations and block ui take +12 seconds to allow to add to cart.

## Expected/actual behavior

When I follow those steps, I see...

Very Slow process to add my variable product to my cart

I was expecting to see...

The Cart Button been blocked and i need wait +12 seconds to add to cart.

I already tried disabling all plugins and leave only store front + woocommerce but the problem continue.

## Isolating the problem

- [x] This bug happens with only WooCommerce plugin active
- [x] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/)
- [x] I can reproduce this bug consistently using the steps above

## WordPress Environment

<details>
```
`
### WordPress Environment ###

Home URL: https://useviper.com.br
Site URL: https://useviper.com.br
WC Version: 3.3.4
Log Directory Writable: ✔
WP Version: 4.9.4
WP Multisite: –
WP Memory Limit: 256 MB
WP Debug Mode: –
WP Cron: ✔
Language: pt_BR

### Server Environment ###

Server Info: Apache
PHP Version: 7.0.28
PHP Post Max Size: 1 GB
PHP Time Limit: 90
PHP Max Input Vars: 4096
cURL Version: 7.58.0
OpenSSL/1.0.2n

SUHOSIN Installed: –
Max Upload Size: 1 GB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 3.3.4
WC Database Prefix: syscoin_
MaxMind GeoIP Database: ✔
Tamanho total do banco de dados: 11.93MB
Tamanho dos dados do banco de dados: 8.46MB
Tamanho do índice do banco de dado: 3.47MB
syscoin_woocommerce_sessions: Dados: 0.22MB + Índices: 0.02MB
syscoin_woocommerce_api_keys: Dados: 0.02MB + Índices: 0.03MB
syscoin_woocommerce_attribute_taxonomies: Dados: 0.02MB + Índices: 0.02MB
syscoin_woocommerce_downloadable_product_permissions: Dados: 0.02MB + Índices: 0.05MB
syscoin_woocommerce_order_items: Dados: 0.02MB + Índices: 0.02MB
syscoin_woocommerce_order_itemmeta: Dados: 0.02MB + Índices: 0.03MB
syscoin_woocommerce_tax_rates: Dados: 0.02MB + Índices: 0.06MB
syscoin_woocommerce_tax_rate_locations: Dados: 0.02MB + Índices: 0.03MB
syscoin_woocommerce_shipping_zones: Dados: 0.02MB + Índices: 0.00MB
syscoin_woocommerce_shipping_zone_locations: Dados: 0.02MB + Índices: 0.03MB
syscoin_woocommerce_shipping_zone_methods: Dados: 0.02MB + Índices: 0.00MB
syscoin_woocommerce_payment_tokens: Dados: 0.02MB + Índices: 0.02MB
syscoin_woocommerce_payment_tokenmeta: Dados: 0.02MB + Índices: 0.03MB
syscoin_woocommerce_log: Dados: 0.02MB + Índices: 0.02MB
syscoin_aryo_activity_log: Dados: 0.41MB + Índices: 0.00MB
syscoin_commentmeta: Dados: 0.00MB + Índices: 0.00MB
syscoin_comments: Dados: 0.00MB + Índices: 0.00MB
syscoin_duplicator_pro_entities: Dados: 0.02MB + Índices: 0.02MB
syscoin_duplicator_pro_packages: Dados: 0.02MB + Índices: 0.02MB
syscoin_email_log: Dados: 1.52MB + Índices: 0.00MB
syscoin_et_social_stats: Dados: 0.02MB + Índices: 0.00MB
syscoin_fpd_categories: Dados: 0.02MB + Índices: 0.00MB
syscoin_fpd_category_products_rel: Dados: 0.02MB + Índices: 0.02MB
syscoin_fpd_products: Dados: 0.02MB + Índices: 0.00MB
syscoin_fpd_views: Dados: 0.02MB + Índices: 0.00MB
syscoin_itsec_lockouts: Dados: 0.02MB + Índices: 0.08MB
syscoin_itsec_log: Dados: 0.02MB + Índices: 0.03MB
syscoin_itsec_logs: Dados: 0.02MB + Índices: 0.09MB
syscoin_itsec_temp: Dados: 0.02MB + Índices: 0.06MB
syscoin_links: Dados: 0.00MB + Índices: 0.00MB
syscoin_options: Dados: 1.49MB + Índices: 0.09MB
syscoin_pmxe_exports: Dados: 0.02MB + Índices: 0.00MB
syscoin_pmxe_posts: Dados: 0.02MB + Índices: 0.00MB
syscoin_pmxe_templates: Dados: 0.02MB + Índices: 0.00MB
syscoin_pmxi_files: Dados: 0.02MB + Índices: 0.00MB
syscoin_pmxi_history: Dados: 0.02MB + Índices: 0.00MB
syscoin_pmxi_imports: Dados: 0.08MB + Índices: 0.00MB
syscoin_pmxi_posts: Dados: 0.02MB + Índices: 0.00MB
syscoin_pmxi_templates: Dados: 0.02MB + Índices: 0.00MB
syscoin_postmeta: Dados: 2.34MB + Índices: 1.61MB
syscoin_posts: Dados: 0.26MB + Índices: 0.11MB
syscoin_rac_abandoncart: Dados: 0.02MB + Índices: 0.00MB
syscoin_rac_email_logs: Dados: 0.02MB + Índices: 0.00MB
syscoin_rac_templates_email: Dados: 0.02MB + Índices: 0.00MB
syscoin_rg_form: Dados: 0.02MB + Índices: 0.00MB
syscoin_rg_form_meta: Dados: 0.02MB + Índices: 0.00MB
syscoin_rg_form_view: Dados: 0.02MB + Índices: 0.03MB
syscoin_rg_incomplete_submissions: Dados: 0.02MB + Índices: 0.02MB
syscoin_rg_lead: Dados: 0.02MB + Índices: 0.03MB
syscoin_rg_lead_detail: Dados: 0.02MB + Índices: 0.06MB
syscoin_rg_lead_detail_long: Dados: 0.02MB + Índices: 0.00MB
syscoin_rg_lead_meta: Dados: 0.02MB + Índices: 0.05MB
syscoin_rg_lead_notes: Dados: 0.02MB + Índices: 0.03MB
syscoin_simple_auction_log: Dados: 0.02MB + Índices: 0.00MB
syscoin_smush_dir_images: Dados: 0.02MB + Índices: 0.03MB
syscoin_snp_log: Dados: 0.02MB + Índices: 0.00MB
syscoin_snp_stats: Dados: 0.02MB + Índices: 0.05MB
syscoin_social_users: Dados: 0.02MB + Índices: 0.02MB
syscoin_stream: Dados: 0.02MB + Índices: 0.13MB
syscoin_stream_meta: Dados: 0.05MB + Índices: 0.08MB
syscoin_subscriptio_scheduled_events: Dados: 0.02MB + Índices: 0.00MB
syscoin_termmeta: Dados: 0.00MB + Índices: 0.01MB
syscoin_terms: Dados: 0.00MB + Índices: 0.01MB
syscoin_term_relationships: Dados: 0.00MB + Índices: 0.00MB
syscoin_term_taxonomy: Dados: 0.00MB + Índices: 0.01MB
syscoin_usermeta: Dados: 0.12MB + Índices: 0.02MB
syscoin_users: Dados: 0.00MB + Índices: 0.01MB
syscoin_vtprd_lifetime_limits_purchaser: Dados: 0.02MB + Índices: 0.02MB
syscoin_vtprd_lifetime_limits_purchaser_logid_rule: Dados: 0.02MB + Índices: 0.02MB
syscoin_vtprd_lifetime_limits_purchaser_rule: Dados: 0.02MB + Índices: 0.02MB
syscoin_vtprd_purchase_log: Dados: 0.02MB + Índices: 0.02MB
syscoin_vtprd_purchase_log_product: Dados: 0.02MB + Índices: 0.02MB
syscoin_vtprd_purchase_log_product_rule: Dados: 0.02MB + Índices: 0.02MB
syscoin_wc_deposits_payment_plans: Dados: 0.02MB + Índices: 0.00MB
syscoin_wc_deposits_payment_plans_schedule: Dados: 0.02MB + Índices: 0.02MB
syscoin_wc_download_log: Dados: 0.02MB + Índices: 0.03MB
syscoin_wc_points_rewards_user_points: Dados: 0.02MB + Índices: 0.03MB
syscoin_wc_points_rewards_user_points_log: Dados: 0.02MB + Índices: 0.05MB
syscoin_wc_point_of_sale_sale_reports: Dados: 0.00MB + Índices: 0.00MB
syscoin_wc_poin_of_sale_grids: Dados: 0.02MB + Índices: 0.00MB
syscoin_wc_poin_of_sale_outlets: Dados: 0.02MB + Índices: 0.00MB
syscoin_wc_poin_of_sale_receipts: Dados: 0.02MB + Índices: 0.00MB
syscoin_wc_poin_of_sale_registers: Dados: 0.02MB + Índices: 0.00MB
syscoin_wc_poin_of_sale_tiles: Dados: 0.02MB + Índices: 0.00MB
syscoin_wc_webhooks: Dados: 0.02MB + Índices: 0.02MB
syscoin_woocommerce_pickup_locations_geodata: Dados: 0.02MB + Índices: 0.05MB
syscoin_woos_search_analytics: Dados: 0.02MB + Índices: 0.00MB
syscoin_wpmelon_advbedit_temp: Dados: 0.05MB + Índices: 0.00MB
syscoin_wpmm_subscribers: Dados: 0.02MB + Índices: 0.00MB
syscoin_yith_wcwl: Dados: 0.02MB + Índices: 0.02MB
syscoin_yith_wcwl_lists: Dados: 0.02MB + Índices: 0.03MB
syscoin_yoast_seo_links: Dados: 0.02MB + Índices: 0.02MB
syscoin_yoast_seo_meta: Dados: 0.42MB + Índices: 0.00MB
syscoin_ywrr_email_blocklist: Dados: 0.02MB + Índices: 0.00MB
syscoin_ywrr_email_schedule: Dados: 0.02MB + Índices: 0.00MB

### Post Type Counts ###

acf: 1
attachment: 77
blocks: 5
custom-css-js: 1
gs-logo-slider: 2
logoshowcase: 1
nav_menu_item: 6
page: 7
pos_temp_register_or: 8
post: 1
product: 2
product_variation: 1296
revision: 6
shop_coupon: 2
vtprd-rule: 1
wc_pickup_location: 1
yith-wccos-ostatus: 3

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ✔

### Active Plugins (83) ###

Gravity Forms: por rocketgenius – 1.9.19
A partir de R$: por SysCoin – 1.0
Add Admin CSS: por Scott Reilly – 1.4
Admin Menu Editor Pro: por Janis Elsts – 2.3
Ninja Popups-Null24.Net: por ArsCode – 4.4.0
Registro de Atividades: por Yakir Sitbon
Maor Chasen
Ariel Klikstein – 2.4.0

iThemes Security: por iThemes – 6.9.2
Bling WooCommerce: por Bling – 1.0.5 – Não testado com a versão ativa do WooCommerce
Custom 404 Pro: por Kunal Nagar – 2.1.0
Simple Custom CSS and JS: por Diana Burduja – 3.15
Date range filter: por Jonathan Bardo
Ricardo Losso – 0.0.11

Desativar Comentários: por Samir Shah – 1.7.1
Duplicate Post: por Enrico Battocchi – 3.2.1
Duplicator Pro - Shared by Null-24.Com: por Snap Creek – 3.4.3
Email Log: por Sudar – 2.2.4
Facebook for WooCommerce: por Facebook – 1.3.2 – Não testado com a versão ativa do WooCommerce
Fancy Product Designer: por Rafael Dery | radykal.de – 3.7.5 – Não testado com a versão ativa do WooCommerce
Gravity PDF: por Gravity PDF – 4.3.2
GS Logo Slider: por Golam Samdani – 1.6
Imsanity: por Shane Bishop – 2.3.10
Instagram Feed Pro Personal: por Smash Balloon – 1.3.1
Tradução (Loco): por Tim Whitlock – 2.0.17
Multistep Product Configurator for WooCommerce: por radykal.de – 1.1.2 – Não testado com a versão ativa do WooCommerce
NCM Products: por Leonardo Miranda – 1.0
Nextend Social Login: por Nextendweb – 3.0.4
Ni WooCommerce Sales Report: por anzia – 2.8.5 – Não testado com a versão ativa do WooCommerce
Particles in login page: por ThemeRose – 1.0
Plugins Garbage Collector: por Vladimir Garagulya – 0.10.3
Recover Abandoned Cart: por Fantastic Plugins – 12.2 – Não testado com a versão ativa do WooCommerce
Really Simple SSL: por Rogier Lankhorst – 2.5.26
Regenerate Thumbnails: por Alex Mills (Viper007Bond) – 3.0.2
Search By SKU - for Woocommerce: por Matthew Lawson – 0.6.1 – Não testado com a versão ativa do WooCommerce
Server IP & Memory Usage Display: por Apasionados
Apasionados del Marketing – 2.1.0

Simple CSS: por Tom Usborne – 1.1
Slate Pro Admin: por Seven Bold – 1.1.3
SparkPost: por SparkPost – 3.1.1
SVG Support: por Benbodhi – 2.3.11
SysCoin Custom Code: por Leonardo Miranda – 1.0
Tawk.to Live Chat: por Tawkto – 0.3.4
Term Management Tools: por scribu – 1.1.4
User Role Editor: por Vladimir Garagulya – 4.40.1
WC Force Pass: por Leo Baiano – 1.0.2 – Não testado com a versão ativa do WooCommerce
Woo Payment Discounts: por WPCodelibrary – 1.0.2
Woocommerce Advanced Bulk Edit: por George Iron – 4.2.2 – Não testado com a versão ativa do WooCommerce
WooCommerce Advanced Free Shipping: por Jeroen Sormani – 1.1.3 – Não testado com a versão ativa do WooCommerce
WooCommerce Auto Restore Stock: por Gerhard Potgieter – 1.0.1 – Não testado com a versão ativa do WooCommerce
WooCommerce Clean Orphaned Variations: por @anttiviljami – 1.0 – Não testado com a versão ativa do WooCommerce
WooCommerce Conversion Tracking: por Tareq Hasan – 2.0 – Não testado com a versão ativa do WooCommerce
WooCommerce Correios: por Claudio Sanches – 3.6.0 – Não testado com a versão ativa do WooCommerce
WooCommerce Cost of Goods: por WooThemes / SkyVerge – 2.2.7 – 2.6.0 está disponível – Não testado com a versão ativa do WooCommerce
WooCommerce Customer History: por Brian Richards – 1.1.1 – 1.2.1 está disponível – Não testado com a versão ativa do WooCommerce
WooCommerce Domination: por Claudio Sanches – 1.1.4 – Não testado com a versão ativa do WooCommerce
WooCommerce Email Customizer: por cxThemes – 3.09 – Não testado com a versão ativa do WooCommerce
WooCommerce Extra Checkout Fields for Brazil: por Claudio Sanches – 3.6.0 – Não testado com a versão ativa do WooCommerce
WooCommerce Gateways Country Limiter: por OnTheGoSystems – 1.4 – Não testado com a versão ativa do WooCommerce
Integração do Google Analytics com o WooCommerce: por WooCommerce – 1.4.3 – Não testado com a versão ativa do WooCommerce
WooCommerce - Gravity Forms Product Add-Ons: por WooThemes – 2.10.10 – 3.2.9 está disponível – Não testado com a versão ativa do WooCommerce
WooCommerce Advanced Product Quantities: por WP BackOffice – 2.1.9 – Não testado com a versão ativa do WooCommerce
WooCommerce MailChimp: por Saint Systems – 2.1.4 – Não testado com a versão ativa do WooCommerce
Woocommerce Order address Print: por Bhavik Patel – 3.2 – Não testado com a versão ativa do WooCommerce
WooCommerce Custom Payment Gateway: por WPRuby – 1.0.6 – Não testado com a versão ativa do WooCommerce
WooCommerce Pagar.me: por Pagar.me
Claudio Sanches – 2.0.11 – Não testado com a versão ativa do WooCommerce

WooCommerce PagSeguro: por Claudio Sanches – 2.12.5 – Não testado com a versão ativa do WooCommerce
WooCommerce Point of Sale: por Actuality Extensions – 4.2.5.14 – Não testado com a versão ativa do WooCommerce
WooCommerce Points and Rewards: por WooThemes – 1.5.13 – 1.6.11 está disponível – Não testado com a versão ativa do WooCommerce
WooCommerce Local Pickup Plus: por SkyVerge – 2.0.1 – 2.3.9 está disponível – Não testado com a versão ativa do WooCommerce
WooCommerce Shortcodes: por WooThemes
Claudio Sanches – 1.0.0 – Não testado com a versão ativa do WooCommerce

WooCommerce Thumbnail Input Quantities: por WP BackOffice – 1.1.2 – Não testado com a versão ativa do WooCommerce
WooCommerce: por Automattic – 3.3.4
WooSidebars: por WooThemes – 1.4.3
Yoast SEO: por Equipe da Yoast – 7.1
WP Change Default From Email: por Subodh Ghulaxe – 1.1.1
WP Fastest Cache: por Emre Vona – 0.8.7.8
WP Logo Showcase Responsive Slider: por WP Online Support – 1.3.5
Manutenção Site: por Designmodo – 2.1.2
WP-Optimize: por David Anderson
Ruhani Rabin
Team Updraft – 2.2.2

WP Smush: por WPMU DEV – 2.7.6
Yoast SEO: WooCommerce: por Time Yoast – 4.9 – Não testado com a versão ativa do WooCommerce
YITH WooCommerce Brands Add-on Premium: por Yithemes – 1.0.9 – Não testado com a versão ativa do WooCommerce
YITH WooCommerce Custom Order Status Premium: por YITHEMES – 1.0.14 – Não testado com a versão ativa do WooCommerce
YITH WooCommerce Recently Viewed Products: por YITHEMES – 1.2.0 – Não testado com a versão ativa do WooCommerce
YITH WooCommerce Review Reminder: por YITHEMES – 1.3.3 – Não testado com a versão ativa do WooCommerce
YITH WooCommerce Wishlist: por YITHEMES – 2.1.2 – Não testado com a versão ativa do WooCommerce

### Settings ###

API Enabled: ✔
Force SSL: –
Currency: BRL (R$)
Currency Position: left
Thousand Separator: .
Decimal Separator: ,
Number of Decimals: 2
Taxonomies: Product Types: auction (auction)
external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)


### WC Pages ###

Base da loja: #4 - /loja/
Carrinho: #5 - /carrinho/
Finalizar compra: #6 - /finalizar-compra/
Minha conta: #7 - /minha-conta/
Termos e condições: ❌ Página não definida

### Theme ###

Name: Flatsome Child
Version: 3.0
Author URL: 
Child Theme: ✔
Parent Theme Name: Flatsome
Parent Theme Version: 3.5.3
Parent Theme Author URL: http://www.uxthemes.com/
WooCommerce Support: ✔

### Templates ###

Overrides: flatsome/woocommerce/archive-product.php
flatsome/woocommerce/cart/cart-empty.php
flatsome/woocommerce/cart/cart.php
flatsome/woocommerce/checkout/form-checkout.php
flatsome/woocommerce/checkout/form-coupon.php
flatsome/woocommerce/checkout/thankyou.php
flatsome/woocommerce/content-product.php
flatsome/woocommerce/content-product_cat.php
flatsome/woocommerce/content-single-product.php
/home/useviper/public_html/wp-content/plugins/woocommerce-email-control/templates/emails/vanilla/admin-cancelled-order.php versão - está desatualizado. A versão do core é 2.5.0
/home/useviper/public_html/wp-content/plugins/woocommerce-email-control/templates/emails/vanilla/admin-failed-order.php versão - está desatualizado. A versão do core é 2.5.0
/home/useviper/public_html/wp-content/plugins/woocommerce-email-control/templates/emails/vanilla/admin-new-order.php versão - está desatualizado. A versão do core é 2.5.0
/home/useviper/public_html/wp-content/plugins/woocommerce-email-control/templates/emails/vanilla/customer-completed-order.php versão - está desatualizado. A versão do core é 2.5.0
/home/useviper/public_html/wp-content/plugins/woocommerce-email-control/templates/emails/vanilla/customer-invoice.php versão - está desatualizado. A versão do core é 3.3.0
/home/useviper/public_html/wp-content/plugins/woocommerce-email-control/templates/emails/vanilla/customer-new-account.php versão - está desatualizado. A versão do core é 1.6.4
/home/useviper/public_html/wp-content/plugins/woocommerce-email-control/templates/emails/vanilla/customer-note.php versão - está desatualizado. A versão do core é 2.5.0
/home/useviper/public_html/wp-content/plugins/woocommerce-email-control/templates/emails/vanilla/customer-on-hold-order.php versão - está desatualizado. A versão do core é 2.5.0
/home/useviper/public_html/wp-content/plugins/woocommerce-email-control/templates/emails/vanilla/customer-processing-order.php versão - está desatualizado. A versão do core é 2.5.0
/home/useviper/public_html/wp-content/plugins/woocommerce-email-control/templates/emails/vanilla/customer-refunded-order.php versão - está desatualizado. A versão do core é 2.5.0
/home/useviper/public_html/wp-content/plugins/woocommerce-email-control/templates/emails/vanilla/customer-reset-password.php versão - está desatualizado. A versão do core é 2.0.0
/home/useviper/public_html/wp-content/plugins/woocommerce-email-control/templates/emails/vanilla/email-addresses.php versão - está desatualizado. A versão do core é 3.2.1
/home/useviper/public_html/wp-content/plugins/woocommerce-email-control/templates/emails/vanilla/email-customer-details.php versão - está desatualizado. A versão do core é 2.5.0
/home/useviper/public_html/wp-content/plugins/woocommerce-email-control/templates/emails/vanilla/email-footer.php versão - está desatualizado. A versão do core é 2.3.0
/home/useviper/public_html/wp-content/plugins/woocommerce-email-control/templates/emails/vanilla/email-header.php versão - está desatualizado. A versão do core é 2.4.0
/home/useviper/public_html/wp-content/plugins/woocommerce-email-control/templates/emails/vanilla/email-order-details.php versão - está desatualizado. A versão do core é 3.3.1
/home/useviper/public_html/wp-content/plugins/woocommerce-email-control/templates/emails/vanilla/email-order-items.php versão - está desatualizado. A versão do core é 3.2.0
/home/useviper/public_html/wp-content/plugins/woocommerce-email-control/templates/emails/vanilla/email-styles.php versão - está desatualizado. A versão do core é 3.3.0
flatsome/woocommerce/global/breadcrumb.php
flatsome/woocommerce/global/quantity-input.php
flatsome/woocommerce/global/wrapper-end.php
flatsome/woocommerce/global/wrapper-start.php
flatsome/woocommerce/loop/loop-end.php
flatsome/woocommerce/loop/loop-start.php
flatsome/woocommerce/loop/orderby.php versão 2.2.0 está desatualizado. A versão do core é 3.3.0
flatsome/woocommerce/loop/pagination.php
flatsome/woocommerce/loop/result-count.php
flatsome/woocommerce/loop/sale-flash.php
flatsome/woocommerce/myaccount/form-login.php
flatsome/woocommerce/myaccount/navigation.php
flatsome/woocommerce/notices/error.php
flatsome/woocommerce/notices/notice.php
flatsome/woocommerce/notices/success.php
flatsome/woocommerce/product-searchform.php
flatsome/woocommerce/single-product/photoswipe.php
flatsome/woocommerce/single-product/price.php
flatsome/woocommerce/single-product/product-image.php
flatsome/woocommerce/single-product/product-thumbnails.php
flatsome/woocommerce/single-product/related.php
flatsome/woocommerce/single-product/review.php
flatsome/woocommerce/single-product/sale-flash.php
flatsome/woocommerce/single-product/share.php
flatsome/woocommerce/single-product/short-description.php
flatsome/woocommerce/single-product/tabs/tabs.php
flatsome/woocommerce/single-product/title.php
flatsome/woocommerce/single-product/up-sells.php
flatsome/woocommerce/single-product-reviews.php
flatsome/woocommerce/single-product.php

`
```
</details>

**Comment by mikejolley** [comment]
I responded in another issue you commented on. https://github.com/woocommerce/woocommerce/issues/12260#issuecomment-378193750

The thing to understand here is that if you have a product with 1000 variations, thats 1000 variations it needs to query, filter, and load when performing cart operations. There is no hard limit on variation number, but we do suggest using no more than 50 in a single product. There are alternatives to variations (see my comment).

Ultimately the product custom tables we're working on will improve this, as could adding indexes to your post meta database table, but this will never work as fast as a simple product and will be slow depending on the complexity of the product.
